### PR TITLE
put together a proposition for better error handling

### PR DESCRIPTION
**Pull request by tarnas14**
This is a proposition - I want to know what you guys think about it, especially regarding handling the backend validation - do we assume that we validate everything in frontend so the validation errors from backend should never happen and we don't handle them (as in: don't worry about showing them in UI)

### CODE REVIEW

**Comment by tarnas14** [comment]
update: added handling for backend schema validation. the (localized) message appears under the input ([example](http://prnt.sc/akoxxr)), but is cleared and overwritten by the frontend validation message on change - if you change the deadline to 'asdf' it will say 'provide valid hour' again

File: .eslintrc.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
   "parserOptions": {
       "ecmaVersion": 6,
       "ecmaFeatures": {
-          "jsx": true
+          "jsx": true,
+          "experimentalObjectRestSpread": true
```
Comment (kjendrzyca) [inline .eslintrc.json:12]: thanks!

File: src/store/ordersActions.js [hunk 2]
```diff
@@ -10,20 +12,43 @@ export function addNewOrder (order) {
...
-        .then(data => {
+        })
+        .then(response => {
+            if (response.status >= 200 && response.status < 300) {
+                return response;
+            }
+
+            const error = new Error(response.statusText);
+            error.apiError = true;
+            error.response = response;
```
Comment (kjendrzyca) [inline src/store/ordersActions.js]: this should be response.json()

File: src/store/ordersActions.js [hunk 3]
```diff
@@ -47,7 +72,7 @@ export function getOrder (id) {
                 dispatch({type: 'GET_ORDER', activeOrder});
             })
             .catch(error => {
-                console.log(error)
+                console.log(error);
```
Comment (kjendrzyca) [inline src/store/ordersActions.js]: please refactor this action to handle errors the same way

**Pull request merged**